datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -286,8 +289,18 @@ def metrics_asserted_pct(self):
             else:
                 return 0
 
+        # If it there have been assertions with at_least=0 the length of the two collections can match
+        # even if there are different metrics in each set
+        not_asserted = self.missing_metrics
+        if not_asserted:
+            return (num_metrics - len(not_asserted)) / num_metrics * 100
+
         return num_asserted / num_metrics * 100
 
+    @property
+    def missing_metrics(self):
```

Comment:
1) Seems there is already a method `not_asserted`: https://github.com/DataDog/integrations-core/blob/0683d52caa42a701457aecea18bb8a5506b39aec/datadog_checks_base/datadog_checks/base/stubs/aggregator.py#L252-L258

2) This would probably have the same effect since if `len(not_asserted)` is 0 then `num_metrics` is same as `num_asserted`

```python
       return (num_metrics - len(not_asserted)) / num_metrics * 100
```

than

```python
        not_asserted = self.missing_metrics
        if not_asserted:
            return (num_metrics - len(not_asserted)) / num_metrics * 100

        return num_asserted / num_metrics * 100
```